NEW Add user permission for create/edit/delete supplier prices

mapiolca opened the pull request:
## Summary

Add an advanced permission "View purchase prices" for the Product module and expose it in the en_US and fr_FR languages.
Allow access to the purchase prices tab when this permission is granted, even without create rights.
Document the new permission in the ChangeLog.

## Testing

php -l htdocs/core/modules/modProduct.class.php
php -l htdocs/core/lib/product.lib.php
php -l htdocs/product/price_suppliers.php

mapiolca committed:
Add copyright notice for Pierre Ardoin

mapiolca committed:
Update permissions for supplier prices management

mapiolca committed:
Add permissions for supplier purchase prices

mapiolca committed:
Enhance supplier price permissions

Added advanced permission checks for supplier price management.

frederic34 commented:
@mapiolca perhaps you can install a plugin "editorconfig" in your favorite code editor to auto configure tabs spaces etc

mapiolca committed:
fix tabs&spaces

mapiolca committed:
fix tab@spaces

mapiolca committed:
Remove unnecessary blank lines in price_suppliers.php

mapiolca commented:
@hregis Ok pour masquer les boutons (j'imagine qu'un if (user a la permission de ) suffit) , mais pour le check, comment je procède ?

hregis commented:
@mapiolca en fait cet ajout suffit car il y a déjà tout ce qu'il faut avec "$usercancreate"

`$usercancreate = (!getDolGlobalString('MAIN_USE_ADVANCED_PERMS') ? $usercancreate : $user->hasRight('product', 'product_advance', 'write_supplier_prices'));`

mapiolca committed:
Remove unauthorized access checks for supplier prices

Removed access control checks for unauthorized supplier price creation and editing.

mapiolca committed:
Remove unnecessary blank line in price_suppliers.php

hregis commented:
@mapiolca si les droits étendues ne sont pas activés "$usercancreate"  sera en fonction du droit de création d'un produit et si les droits étendus sont activés "$usercancreate" sera en fonction du droit de créer un prix fournisseur

mapiolca committed:
Refactor user permission checks for supplier prices

replace $usercanwritesupplierprice with $usercancreate

mapiolca committed:
Refactor edit and remove links for supplier prices

mapiolca committed:
Merge branch 'develop' into patch-65

mapiolca committed:
Merge branch 'develop' into patch-65

eldy commented on htdocs/product/price_suppliers.php:
```diff
@@ -348,7 +349,7 @@
 						$ret = $object->setSupplierPriceExpression($price_expression);
 						if ($ret < 0) {
-							$error++;
+							$error++;$
```
Oops char $ not expected here

mapiolca committed:
Update price_suppliers.php

mapiolca committed:
Merge branch 'develop' into patch-65
